third_party/intel/lib/TritonIntelGPUToLLVM/DecomposeUnsupportedConversions.cpp

```diff
@@ -14,136 +16,18 @@ namespace mlir::triton::gpu::intel {
-      auto srcType = cast<RankedTensorType>(cvtOp.getSrc().getType());
-      auto dstType = cast<RankedTensorType>(cvtOp.getType());
-      auto srcDpas =
-          dyn_cast<triton::gpu::intel::DpasEncodingAttr>(srcType.getEncoding());
-      auto dstDotOp =
-          dyn_cast<triton::gpu::DotOperandEncodingAttr>(dstType.getEncoding());
-      if (srcDpas && dstDotOp &&
-          !triton::gpu::intel::isDpasToDotShortcut(srcType, dstType)) {
-        auto tmpType = RankedTensorType::get(
-            dstType.getShape(), dstType.getElementType(),
-            triton::gpu::BlockedEncodingAttr::get(
-                mod.getContext(), srcType.getShape(), getSizePerThread(srcDpas),
-                getOrder(srcDpas), numWarps, threadsPerWarp, numCTAs));
-        auto tmp = builder.create<triton::gpu::ConvertLayoutOp>(
-            cvtOp.getLoc(), tmpType, cvtOp.getSrc());
-        addAttrs(tmp, cvtOp->getAttrs());
-        auto newConvert = builder.create<triton::gpu::ConvertLayoutOp>(
-            cvtOp.getLoc(), dstType, tmp);
-        addAttrs(newConvert, cvtOp->getAttrs());
-        cvtOp.replaceAllUsesWith(newConvert.getResult());
-        cvtOp.erase();
-      }
-    });
-    /* -------------------------------- */
-    // Replace `blocked -> dot_op` with `blocked -> shared -> dot_op`
-    // because the codegen doesn't handle `blocked -> dot_op` directly
-    /* -------------------------------- */
-    mod.walk([&](triton::gpu::ConvertLayoutOp cvtOp) -> void {
-      OpBuilder builder(cvtOp);
-      auto srcType = cast<RankedTensorType>(cvtOp.getSrc().getType());
-      auto dstType = cast<RankedTensorType>(cvtOp.getType());
-      auto srcBlocked =
-          dyn_cast<triton::gpu::BlockedEncodingAttr>(srcType.getEncoding());
-      auto dstDotOp =
-          dyn_cast<triton::gpu::DotOperandEncodingAttr>(dstType.getEncoding());
-      if (srcBlocked && dstDotOp) {
-        Attribute sharedMemorySpace =
-            triton::gpu::SharedMemorySpaceAttr::get(srcType.getContext());
-        auto tmpType = triton::MemDescType::get(
-            dstType.getShape(), dstType.getElementType(),
-            triton::gpu::SharedEncodingAttr::get(
-                mod.getContext(), dstDotOp, srcType.getShape(),
-                srcBlocked.getOrder(), srcBlocked.getCTALayout(),
-                srcType.getElementType()),
-            sharedMemorySpace);
-        auto tmp = builder.create<triton::gpu::LocalAllocOp>(
-            cvtOp.getLoc(), tmpType, cvtOp.getSrc());
-        auto newConvert = builder.create<triton::gpu::LocalLoadOp>(
-            cvtOp.getLoc(), dstType, tmp);
-        cvtOp.replaceAllUsesWith(newConvert.getResult());
-        cvtOp.erase();
-      }
-    });
+    triton::gpu::decomposeSplatOpToSharedLayoutConversion(mod);
+
+    triton::gpu::decomposeTensorCoreToDotLayoutConversion(mod,
+                                                          isDpasToDotShortcut);
+
+    triton::gpu::decomposeBlockedToDotLayoutConversion(mod);
```

Comment:
[nit] to be consistent with NVidia version.
```suggestion
    triton::gpu::decomposeSplatOpToSharedLayoutConversion(mod);
    triton::gpu::decomposeTensorCoreToDotLayoutConversion(mod,
                                                          isDpasToDotShortcut);
    triton::gpu::decomposeBlockedToDotLayoutConversion(mod);
```